core/src/main/java/io/fabric8/maven/core/config/Profile.java

```diff
@@ -94,13 +102,30 @@ public ProcessorConfig getGeneratorConfig() {
         return generatorConfig;
     }
 
-    public int getOrder() { return order; }
-
-    public static class OrderComparator implements Comparator<Profile> {
+    public int getOrder() {
+        return order;
+    }
 
-        @Override
-        public int compare(Profile o1, Profile o2) {
-            return o1.order - o2.order;
+    @Override
+    public int compareTo(Profile o) {
```

Comment:
@jimmidyson could you give this method a quick look ? It tries hard to determine a reproducible order, if the profiles are not equal (equals in the sense of ==). Don't know whether the creationTime thing is needed, since its arbitrary then anyway (the "name" is always the same when comparing since only same named profiles are compared).

Maybe going with the hashcode() is simpler.